packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -232,7 +231,12 @@ export const makeZCFZygote = async (
   }
 
   start.length <= 3 || Fail`invalid start parameters`;
-  const takesBaggage = start.length === 3;
+  const durabilityRequired = ['canBeUpgraded', 'canUpgrade'].includes(
+    checks.upgradability,
+  );
+
+  const privateArgsShape = checks.privateArgs || rest.privateArgsShape;
+  const customTermsShape = checks.customTerms || rest.customTermsShape;
```

Comment:
I think I would require them to be in `checks` rather than allow them to be in one of two places.  I would just drop `rest`.